wcomponents-theme/src/test/intern/wc.file.util.test.js

```diff
@@ -0,0 +1,98 @@
+define(["intern!object", "intern/chai!assert", "./resources/test.utils!"],
+	function (registerSuite, assert, testutils) {
+		"use strict";
+		var fileUtil,
+			wcconfig;
+		registerSuite({
+			name: "wc/file/util",
+			setup: function () {
+				return testutils.setupHelper(["wc/file/util", "wc/config"], function (util, config) {
+					fileUtil = util;
+					wcconfig = config;
+				});
+			},
+			testBlobToFile: function () {
+				var expectedMime = "text/plain",
+					expectedFile = "test.file",
+					content = "This is my blob content",
+					blob = new Blob([content], {type: expectedMime}),
+					file = fileUtil.blobToFile(blob, {
+						name: expectedFile
+					});
+				assert.strictEqual(file.name, expectedFile);
+				assert.strictEqual(file.type, expectedMime);
+				assert.strictEqual(file.size, content.length);
+				// https://developer.mozilla.org/en-US/docs/Web/API/File#Properties
+				assert.isDefined(file.lastModified, "Check this is a file (not a blob).");
+				assert.strictEqual(file.lastModified, file.lastModifiedDate.getTime());
+				return new Promise(function (win) {
+					var fileReader = new FileReader();
+					fileReader.onload = function () {
+						assert.strictEqual(fileReader.result, content);
+						assert.notStrictEqual(fileReader.result, "content");
+						win();
+					};
+					fileReader.readAsText(file);
+				});
+			},
+			testDataURItoBlob: function () {
+				var expectedText = "hello world",
+					expectedMime = "text/plain",
+					dataurl = 'data:' + expectedMime + ';base64,' + btoa(expectedText),
+					blob;
+
+				blob = fileUtil.dataURItoBlob(dataurl);
+				assert.strictEqual(blob.type, expectedMime);
+				assert.strictEqual(blob.size, expectedText.length);
```

Comment:
Hmmm interesting one this one. I'm pretty sure it's "wrong" however it will probably work forever.
The size of the blob is in bytes. Your test assumes that each character corresponds to one byte and since your test string is in the ASCII range it _probably_ always will be (I'm honestly not sure).

It certainly would break if you put a character in there outside the ASCII range. You can change the string to:

`expectedText = "hello world\u00DC"`

In trying this you will discover that `btoa` is broken, so you would need to add this [b64EncodeUnicode](https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_Unicode_Problem) function to the test and call it instead of btoa `dataurl = 'data:' + expectedMime + ';base64,' + b64EncodeUnicode(expectedText),`

Then you will see what I mean, this assertion will fail. Perhaps you should simply assert that the size is greater that the length of the expected text...?

Reply:
wow, great find, yes agree